rush-plugins/rush-resolver-cache-plugin/src/externals.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type Module from 'node:module';
+
+import type { Operation as OperationType, OperationStatus as OperationStatusType } from '@rushstack/rush-sdk';
+import type { PnpmShrinkwrapFile as PnpmShrinkwrapFileType } from '@rushstack/rush-sdk/lib/logic/pnpm/PnpmShrinkwrapFile';
+import type * as rushSdkType from '@rushstack/rush-sdk';
+
+// Ultra-cheap "I am a Rush plugin" import of rush-lib
+// eslint-disable-next-line @typescript-eslint/naming-convention
+declare const ___rush___rushLibModule: typeof rushSdkType;
```

Comment:
What's wrong with just importing from `rush-sdk`? This is far more fragile.

Reply:
Direct imports from `@rushstack/rush-sdk` massively inflate the code footprint, by loading an entirely separate and unnecessary copy of `@rushstack/node-core-library`.

I'd love to not need these, but that involves a significant refactor of `@rushstack/rush-sdk` to allow different entry points depending on the nature of the caller.